Lib/importlib/_bootstrap_external.py

```diff
@@ -211,10 +212,10 @@ def _write_atomic(path, data, mode=0o666):
         with _io.FileIO(fd, 'wb') as file:
             file.write(data)
         _os.replace(path_tmp, path)
-    except OSError:
+    except (OSError, ValueError):
```

Comment:
How can it raise ValueError? If `_os.open()` has not failed, `path_tmp` is valid, and `path` should be valid as well. 

On other hand, I suggest to remove the temporary file on any error (including KeyboardInterrupt and MemoryError). A bare `except` is safe here, because the exception is re-raised.

```suggestion
    except:
```

Reply:
Yes, you're right. I remember that I was like "mmh, writing should be safe, right?". I'll update the PR (and check the tests).